code/modules/mob/living/brain/say.dm

```diff
@@ -1,6 +1,10 @@
 /mob/living/brain/say(message, language)
+	var/obj/item/mmi/posibrain/P = container
```

Comment:
replace all of this with a call back to the "container"

Reply:
What do you mean?

Reply:
if there is a container var just call some proc on that container like container.can_say() or similar so all the container specific code is contained in the brain container